promql/functions.go

```diff
@@ -814,11 +863,11 @@ var functions = map[string]*Function{
 		Call:       funcResets,
 	},
 	"round": {
-		Name:         "round",
-		ArgTypes:     []model.ValueType{model.ValVector, model.ValScalar},
-		OptionalArgs: 1,
-		ReturnType:   model.ValVector,
-		Call:         funcRound,
+		Name:       "round",
+		ArgTypes:   []model.ValueType{model.ValVector, model.ValScalar},
+		Variadic:   true,
```

Comment:
Seems to be the best compromise.

`Variadic int`, -1 for limitless, other for exactly `i` optional args, meaning either all `i` are there or none (at least it cannot hurt to pick the most restrictive version for now).